src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefixAnalyzer.cs

```diff
@@ -450,64 +449,76 @@ static bool TryFindFixedSets(RegexNode node, List<RegexFindOptimizations.FixedDi
                 int s1RangeLength = s1.Range is not null ? GetRangeLength(s1.Range.Value, s1Negated) : 0;
                 int s2RangeLength = s2.Range is not null ? GetRangeLength(s2.Range.Value, s2Negated) : 0;
 
-                Debug.Assert(!s1Negated || s1Chars is null);
-                Debug.Assert(!s2Negated || s2Chars is null);
+                // If one set is negated and the other isn't, prefer the non-negated set.
+                if (s1Negated != s2Negated)
+                {
+                    return s2Negated ? -1 : 1;
+                }
 
-                // If both have chars, prioritize the one with the smaller frequency for those chars.
-                if (s1Chars is not null && s2Chars is not null)
+                // If we extracted only a few chars and the sets are negated, they both represent very large
+                // sets that are difficult to compare for quality.
+                if (!s1Negated)
                 {
-                    // Prefer sets with less frequent values.  The frequency is only an approximation,
-                    // used as a tie-breaker when we'd otherwise effectively be picking randomly.
-                    // True frequencies will vary widely based on the actual data being searched, the language of the data, etc.
-                    float s1Frequency = SumFrequencies(s1Chars);
-                    float s2Frequency = SumFrequencies(s2Chars);
+                    Debug.Assert(!s2Negated);
 
-                    if (s1Frequency != s2Frequency)
+                    // If both have chars, prioritize the one with the smaller frequency for those chars.
+                    if (s1Chars is not null && s2Chars is not null)
                     {
-                        return s1Frequency.CompareTo(s2Frequency);
-                    }
+                        Debug.Assert(!s2Negated);
```

Comment:
Why is this needed if you have the same one above?

Reply:
Heh, it's not.  I think I intended to move it and instead copied it.